google-cloud-spanner/lib/google/cloud/spanner/results.rb

```diff
@@ -143,28 +157,25 @@ def rows
                 # Flush the buffered responses now that they are all handled
                 buffered_responses = []
               end
-            rescue GRPC::Cancelled, GRPC::DeadlineExceeded, GRPC::Internal,
-                   GRPC::ResourceExhausted, GRPC::Unauthenticated,
-                   GRPC::Unavailable, GRPC::Core::CallError => err
-              if resume_token.nil? || resume_token.empty?
-                # Re-raise if the resume_token is not a valid value.
-                # This can happen if the buffer was flushed.
-                raise Google::Cloud::Error.from_error(err)
-              end
+            rescue GRPC::Core::CallError,
+              Google::Cloud::AbortedError,
+              Google::Cloud::CanceledError,
+              Google::Cloud::DeadlineExceededError,
+              Google::Cloud::InternalError,
+              Google::Cloud::ResourceExhaustedError,
+              Google::Cloud::UnauthenticatedError,
+              Google::Cloud::UnavailableError => err
 
-              # Resume the stream from the last known resume_token
-              if @execute_query_options
-                @enum = @service.execute_streaming_sql \
-                  @session_path, @sql,
-                  @execute_query_options.merge(resume_token: resume_token)
+              if resumable?(resume_token)
+                should_resume_request = true
+              elsif retryable?(err)
+                should_retry_request = true
+              elsif err.is_a?(Google::Cloud::Error)
+                raise err
               else
-                @enum = @service.streaming_read_table \
-                  @session_path, @table, @columns,
-                  @read_options.merge(resume_token: resume_token)
+                raise Google::Cloud::Error.from_error(err)
               end
 
-              # Flush the buffered responses to reset to the resume_token
-              buffered_responses = []
             rescue GRPC::BadStatus => err
               raise Google::Cloud::Error.from_error(err)
```

Comment:
@thiagotnunes  `rescue GRPC::BadStatus`  need to remove. Generator always going to throw `Google::Cloud::Error` so no need to capture again and cast.

Reply:
We will need to keep this for the time being (refer to the discussion below).